Realm/Tests/Swift/SwiftSchemaTests.swift

```diff
@@ -222,6 +224,11 @@ class SwiftRLMSchemaTests: RLMMultiProcessTestCase {
     }
 
     func testAccessSchemaCreatesObjectWhichAttempsInsertionsToArrayProperty() {
+        // This test depends on InitAppendsToArrayValue having a nil _objectSchema.
+        // There is a chicken and egg race problem here as InitAppendsToArrayValue may
+        // have its object schema registered before InitAppendsToArrayProperty.
```

Comment:
This again is a test revealing a bug. `InitAppendsToArrayValue` must have a nil `_objectSchema` at the point where the getter is called because the getter needs to be called before we can construct the ObjectSchema for that type. If it's non-nil, that means we assigned a garbage value to the ivar.

Reply:
The strange thing about this test is that it will pass when running the test suite in the Xcode project / swift package from Xcode. But it will fail when run from `build.sh test-swiftpm`

Reply:
I don't see any failures with `test-swiftpm` or `test-swiftpm-address` when I remove the cleanup code, and I don't really see how extra classes could be producing failures there.

Reply:
hmm I also no longer see it

Reply:
I can get it to fail when I disable the sync tests in the swift package and then run it with the asan.